File: app/src/main/java/workshop/akbolatss/tagsnews/di/module/NewsListModule.java
package workshop.akbolatss.tagsnews.di.module;

import dagger.Module;
import dagger.Provides;
import workshop.akbolatss.tagsnews.di.scope.ActivityScope;
import workshop.akbolatss.tagsnews.screen.news.NewsView;


@Module
public class NewsListModule {

    private NewsView mView;

    public NewsListModule(NewsView mView) {
        this.mView = mView;
    }

    @ActivityScope
    @Provides
    NewsView provideView(){
        return mView;
    }
}

File: app/src/main/java/workshop/akbolatss/tagsnews/di/module/NewsListModule.kt
package workshop.akbolatss.tagsnews.di.module

import dagger.Module
import dagger.Provides
import workshop.akbolatss.tagsnews.di.scope.ActivityScope
import workshop.akbolatss.tagsnews.screen.news.NewsView

/**
 * Dagger2 module for NewsComponent
 * @see workshop.akbolatss.tagsnews.di.component.NewsComponent
 */
@Module
class NewsListModule(private val mView: NewsView) {

    /**
     * MVP View Provider
     * @see NewsView
     */
    @ActivityScope
    @Provides
    internal fun provideView(): NewsView {
        return mView
    }
}
